src/cyw43_ctrl.c

```diff
@@ -401,13 +405,22 @@ void cyw43_cb_process_async_event(void *cb_data, const cyw43_async_event_t *ev)
         } else {
             // PSK_SUP failure
             self->wifi_join_state = WIFI_JOIN_STATE_BADAUTH;
+            connect_done = true;
         }
     }
 
     if (self->wifi_join_state == WIFI_JOIN_STATE_ALL) {
+
         // STA connected
         self->wifi_join_state = WIFI_JOIN_STATE_ACTIVE;
         cyw43_cb_tcpip_set_link_up(self, CYW43_ITF_STA);
+        connect_done = true;
+    }
+
+    // Restore power management value
+    if (connect_done && self->saved_pm) {
```

Comment:
I think instead of `connect_done` you can check `(wifi_join_state & WIFI_JOIN_STATE_KIND_MASK) != 0`

Reply:
Won't that cause this line to get executed every poll - because WIFI_JOIN_STATE_ACTIVE is set?

Reply:
Actually - ignore that, saved_pm won't be set. I'll fix it as you suggest.